src/main/java/mate/academy/spring/controller/BookController.java

```diff
@@ -19,14 +23,31 @@
     private BookService bookService;
 
     @GetMapping("/all")
-    public String getAllBooks(ModelMap model) {
+    public String getAllBooks(Model model) {
         List<Book> books = bookService.getAll();
-        model.put("books", books);
+        model.addAttribute("books", books);
         return "books";
     }
 
-    @GetMapping("/info")
-    public String bookInfo(@RequestParam("book_id") Long id, Model model) {
+    @GetMapping("/add")
+    public ModelAndView showAddBookForm() {
+        return new ModelAndView("createBook", "bookForm", new Book());
+    }
+
+    @PostMapping("/add")
+    public String addBook(
+            @ModelAttribute("bookForm") Book book, BindingResult result, Model model) {
+        if (result.hasErrors()) {
+            model.addAttribute("message", "Book creating error");
+            return "errorPage";
+        }
+        bookService.add(book);
+        model.addAttribute("books", bookService.getAll());
```

Comment:
we can call method `getAllBooks(model)` here

Reply:
+